tests/different-conditions-fold/Makefile

```
@@ -0,0 +1,38 @@
+include $(dir $(MAKEFILE_LIST))../common.mk
+
+all:
+	mkdir -p build1 build2
+	cd build1 ; cat ../main_hdr.c ../main1.c ../main_ftr.c > ../main.c ; $(CC) $(CXXFLAGS) ../main.c -o testcase
+	cd build2 ; cat ../main_hdr.c ../main2.c ../main_ftr.c > ../main.c ; $(CC) $(CXXFLAGS) ../main.c -o testcase
+
+run: json txt lcov cobertura html sonarqube coveralls
+
+coverage.json:
+	cd build1 ; ./testcase
+	cd build2 ; ./testcase
+	$(GCOVR) --verbose --merge-mode-conditions=fold --json-pretty --json coverage.json
+
+json: coverage.json
+	$(GCOVR) --verbose --json-add-tracefile $< > coverage.log 2>&1 ; test $$? -eq 0 | cat coverage.log
```

Would first check strict and test for the expected exit code and then check the redirected log file. 
```suggestion
	cd build1 ; ./testcase
	cd build2 ; ./testcase
	$(GCOVR) --verbose --merge-mode-conditions=fold --json-pretty --json coverage.json

json: coverage.json
```

---

So combine the 2 tests into 1 test and check failing first (strict), then check that the merge is good (fold)?

---

I'm not sure I understand what you are suggesting here.

---

Sorry, the comment was old. I would not use `--merge-mode-conditions=fold` to check if the value `fold` is used as default.

---

Also the second gcovr call isn't needed.

---

So delete all the other formats from the fold test except generating the coverage.json file? Sorry again, just trying to make sure I get what you are asking, but I'm not sure what you mean with `I would not use --merge-mode-conditions=fold to check if the value fold is used as default.`.

Also, what can I do to resolve the test issues which seem to be from environment differences? Like the mac test?

The gcc-14 one I'm not sure I get what differs and why there's variation in the data.

---

Have you used docker for testing? The GCC data can be generated for all compilers with the docker by running the command `nox -rs 'docker_run_compiler(all)' -- --s tests -- --update_reference` (I need to update the documentation here).
If there are differences for Windows or MacOs the differences are uploaded to the pipeline run. You can download them, put them to the root of your clone and run `not -rs import_reference -- *.zip`.